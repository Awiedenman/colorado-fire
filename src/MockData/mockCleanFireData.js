/* eslint-disable */
export const mockCleanFireData = [{
  fire_name: "416",
  acres_burned: " 54129 acres",
  image: undefined,
  last_update: " 7/24/2018 10:15:00 PM",
  latitude: "37.4605800000001",
  longitude: "-107.8081",
  verified: true
}]