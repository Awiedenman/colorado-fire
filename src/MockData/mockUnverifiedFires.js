 export const mockFiresFromDb = [{
   id: 10,
   first_name: "",
   last_name: "will's blaze",
   fire_name: null,
   image: null,
   acres_burned: null,
   last_update: null,
   latitude: "40",
   longitude: "-79",
   city: "",
   state: "",
   zip_code: "",
   verified: false,
   created_at: "2018-07-31T23:04:57.695Z",
   updated_at: "2018-07-31T23:04:57.695Z"
 }];