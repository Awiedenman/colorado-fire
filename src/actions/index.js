export const storeCurrentFireData = (cleanedCurrentFireData) => ({
  type: 'STORE_CURRENT_FIRE_DATA',
  cleanedCurrentFireData
});