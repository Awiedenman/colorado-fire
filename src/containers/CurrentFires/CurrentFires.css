.gm-style-pbc {
  width: 40%;
}