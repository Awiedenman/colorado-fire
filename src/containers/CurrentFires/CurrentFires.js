import React, { Component } from 'react';
import { apiKey } from '../../api-key';
import { Map, InfoWindow, Marker, GoogleApiWrapper } from 'google-maps-react';
import { connect } from 'react-redux';


export class CurrentFires extends Component {
  
  render() {
    const currentFireMarkers = this.props.currentFires.map(fire => {
      return <Marker
          title={'The marker`s title will appear as a fire.'}
          name={fire.name}
          position={{lat: fire.latitude, lng: fire.longitude }}
          key={fire.name} 
        />
    })
      
    const initialCenter = {
      lat: 39,
      lng: -96.9
    }
    
    const style = {
      width: '100%',
      height: '100%'
    }
    
    return(
      <Map 
        style={style}
        google={this.props.google} 
        initialCenter={initialCenter}
        center={initialCenter}
        zoom= {5}
      >
      {currentFireMarkers}
        <InfoWindow onClose={this.onInfoWindowClose}>
            <div>
              {/* <h1>{this.state.selectedPlace.name}</h1> */}
            </div>
        </InfoWindow>
      </Map>
   )
  }
}

const mapStateToProps = (state) => ({
  currentFires: state.currentFires
})

const googleWrapper = GoogleApiWrapper({ apiKey: apiKey })(CurrentFires)

export default connect(mapStateToProps)(googleWrapper);

