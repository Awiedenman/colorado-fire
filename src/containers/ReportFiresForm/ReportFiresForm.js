import React, { Component } from 'react';
// import { postUnverifiedFires } from '../../heplers/apiCalls/apiCalls';

export class ReportFiresForm extends Component {
  constructor(props){
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      latitude: '',
      longitude: '',
      city: '',
      state:'',
      zipCode: ''
    }
  }

  handleChange = (e) =>{
    const { value, name } = e.target;
    this.setState({[name]: value})
  }

  // postUnverifiedFires = (e) => {
  //   e.preventDefault();
  //   postUnverifiedFires(this.state)
  // }

  render() {
    return (
      <form>
        <section className='reportFire-form-section'>
        <h3>Name</h3>
          <label for="name"></label>
            <input 
              onChange={this.handleChange} 
              name='firstName' 
              value={this.state.firstName}
              id='name'
              placeholder='First Name'
            />
          <label for="lastName"></label>
            <input 
              onChange={this.handleChange} 
              name='lastName' 
              value={this.state.lastName}
              id='lastName'
              placeholder='Last Name'
            />
        </section>
        <br />
        <section className='reportFire-form-section'>
          <h3>(Optional)</h3>
          <label for="latitude"></label>
            <input 
              onChange={this.handleChange} 
              name='latitude' 
              value={this.state.latitude}
              id='latitude'
              placeholder='Latitude'
            />
          <label for="longitude"></label>
            <input 
              onChange={this.handleChange} 
              name='longitude' 
              value={this.state.longitude}
              id='longitude'
              placeholder='Longitude'
            />
        </section>
        <br />
        <section className='reportFire-form-section'>
          <label for="city"></label>
            <h3>Location</h3>
            <input 
              onChange={this.handleChange} 
              name='city' 
              value={this.state.city}
              id='city'
              placeholder='City'
            />
          <label for="state"></label>
            <input 
              onChange={this.handleChange} 
              name='state' 
              value={this.state.state}
              id='state'
              placeholder='State'
            />
          <label for="zip code"></label>
            <input 
              onChange={this.handleChange} 
              name='zipCode' 
              value={this.state.zipCode}
              id='zipCode'
              placeholder='Zip Code'
            />  
        </section>
        {/* <button onClick={postUnverifiedFires}>Submit</button> */}
      </form>
    )
  }
}
