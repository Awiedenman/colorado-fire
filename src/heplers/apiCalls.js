var convert = require('xml-js');
 
 export const currentFireRequest = async () => {
   const url = 'https://cors-anywhere.herokuapp.com/https://www.geomac.gov/DynContent/georss/nifc_large_firesW3C.xml';
   try{
     const xmlResponse = await fetch(url);
     const currentFireDataXml = await xmlResponse.text();
     var jsonData = convert.xml2json(
       currentFireDataXml, {
         compact: true,
         spaces: 2
        });
        // console.log(currentFireDataXml)
        // console.log(jsonData)
        return jsonData
      } catch(error){
        throw Error('Couldn\'t retreive the current fire')
      }
 }