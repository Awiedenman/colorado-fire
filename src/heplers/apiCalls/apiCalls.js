var convert = require('xml-js');
 
 export const currentFireRequest = async () => {
   const url = 'https://cors-anywhere.herokuapp.com/https://www.geomac.gov/DynContent/georss/nifc_large_firesW3C.xml';
   try{
     const xmlResponse = await fetch(url);
     const currentFireDataXml = await xmlResponse.text();
     var jsonData = convert.xml2json(
       currentFireDataXml, {
         compact: true,
         spaces: 2
        });
        return jsonData
      } catch(error){
        throw Error('Couldn\'t retreive the current fires list')
      }
 }

 export const currentFireArticlesRequest = async () => {
   const url = 'https://cors-anywhere.herokuapp.com/http://inciweb.nwcg.gov/feeds/rss/articles/';
   try {
     const xmlResponse = await fetch(url);
     const currentFireArticlesXml = await xmlResponse.text();
     var jsonData = convert.xml2json(
       currentFireArticlesXml, {
         compact: true,
         spaces: 2
       });
    //  console.log(currentFireArticlesXml)
    //  console.log(jsonData)
     return jsonData
   } catch (error) {
     throw Error('Couldn\'t retreive the current fire')
   }
 }