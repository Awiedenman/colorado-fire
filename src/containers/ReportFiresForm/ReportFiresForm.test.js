import React from 'react';
import { shallow } from 'enzyme';
import { ReportFiresForm, mapDispatchToProps } from './ReportFiresForm';
import { postUnverifiedFires } from '../../heplers/apiCalls/apiCalls';
import { mockFiresFromDb } from '../../MockData/mockUnverifiedFires';
import { mapStateToProps } from '../CurrentFires/CurrentFires';
import { addUnverifiedFire } from '../../actions/index';

describe('ReportFiresForm', () => {
  describe('handleChange', () => {
    test('should change the state on input', () => {
      const wrapper = shallow(<ReportFiresForm/>);

      const mockEvent = { target: { name: 'firstName', value : 'Austin'}};

      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state('firstName')).toEqual(mockEvent.target.value);
    });
  });

  describe('handlePostUnverifiedFires', () => {
    test('should call fetch', async () => {
      window.fetch = jest.fn().mockImplementation(() =>
        Promise.resolve({
          ok: true,
          json: () => Promise.resolve({
            city:"",
            first_name:"",
            id:11,
            last_name:"",
            latitude:"",
            longitude:"",
            state:"",
            verified: false,
            zip_code:""
          })
        }));

      await postUnverifiedFires('Denver', 'austin', '11', 'Wiedenman', '45', '50', 'CO', false, 80219);

      expect(window.fetch).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    test('should call dispatch when addUnverifiedFire is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addUnverifiedFire(mockFiresFromDb)
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.addUnverifiedFire(mockFiresFromDb);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
  
  test('should match snapshot', () => {
    const wrapper = shallow( < ReportFiresForm /> );

    expect(wrapper).toMatchSnapshot();
  });
});


  
  

  

