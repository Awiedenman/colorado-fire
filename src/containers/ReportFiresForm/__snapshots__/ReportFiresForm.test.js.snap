// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportFiresForm should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportFiresForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="form-fire-logo"
/>,
        <h2>
          Report a Fire
</h2>,
        <section
          className="reportFire-form-section"
>
          <label
                    for="name"
          />
          <input
                    id="name"
                    name="first_name"
                    onChange={[Function]}
                    placeholder="First Name"
                    value=""
          />
          <label
                    for="lastName"
          />
          <input
                    id="lastName"
                    name="last_name"
                    onChange={[Function]}
                    placeholder="Last Name"
                    value=""
          />
</section>,
        <br />,
        <section
          className="reportFire-form-section"
>
          <h3>
                    <span
                              className="star"
                    >
                              *
                    </span>
                    Required
          </h3>
          <label
                    for="latitude"
          />
          <input
                    id="latitude"
                    name="latitude"
                    onChange={[Function]}
                    placeholder="Latitude"
                    required={true}
                    value=""
          />
          <label
                    for="longitude"
          />
          <input
                    id="longitude"
                    name="longitude"
                    onChange={[Function]}
                    placeholder="Longitude"
                    required={true}
                    value=""
          />
</section>,
        <br />,
        <section
          className="reportFire-form-section"
>
          <label
                    for="city"
          />
          <br />
          <h3>
                    <span
                              className="star"
                    >
                              *
                    </span>
                     optional
          </h3>
          <input
                    id="city"
                    name="city"
                    onChange={[Function]}
                    placeholder="City"
                    value=""
          />
          <label
                    for="state"
          />
          <input
                    id="state"
                    name="state"
                    onChange={[Function]}
                    placeholder="State"
                    value=""
          />
          <label
                    for="zip code"
          />
          <input
                    id="zipCode"
                    name="zip_code"
                    onChange={[Function]}
                    placeholder="Zip Code"
                    value=""
          />
</section>,
        <button
          onClick={[Function]}
>
          Submit
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-fire-logo",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Report a Fire",
        },
        "ref": null,
        "rendered": "Report a Fire",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              for="name"
/>,
            <input
              id="name"
              name="first_name"
              onChange={[Function]}
              placeholder="First Name"
              value=""
/>,
            <label
              for="lastName"
/>,
            <input
              id="lastName"
              name="last_name"
              onChange={[Function]}
              placeholder="Last Name"
              value=""
/>,
          ],
          "className": "reportFire-form-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "for": "name",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "name",
              "name": "first_name",
              "onChange": [Function],
              "placeholder": "First Name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "for": "lastName",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "lastName",
              "name": "last_name",
              "onChange": [Function],
              "placeholder": "Last Name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              <span
                            className="star"
              >
                            *
              </span>
              Required
</h3>,
            <label
              for="latitude"
/>,
            <input
              id="latitude"
              name="latitude"
              onChange={[Function]}
              placeholder="Latitude"
              required={true}
              value=""
/>,
            <label
              for="longitude"
/>,
            <input
              id="longitude"
              name="longitude"
              onChange={[Function]}
              placeholder="Longitude"
              required={true}
              value=""
/>,
          ],
          "className": "reportFire-form-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="star"
>
                  *
</span>,
                "Required",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "*",
                  "className": "star",
                },
                "ref": null,
                "rendered": "*",
                "type": "span",
              },
              "Required",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "for": "latitude",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "latitude",
              "name": "latitude",
              "onChange": [Function],
              "placeholder": "Latitude",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "for": "longitude",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "longitude",
              "name": "longitude",
              "onChange": [Function],
              "placeholder": "Longitude",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              for="city"
/>,
            <br />,
            <h3>
              <span
                            className="star"
              >
                            *
              </span>
               optional
</h3>,
            <input
              id="city"
              name="city"
              onChange={[Function]}
              placeholder="City"
              value=""
/>,
            <label
              for="state"
/>,
            <input
              id="state"
              name="state"
              onChange={[Function]}
              placeholder="State"
              value=""
/>,
            <label
              for="zip code"
/>,
            <input
              id="zipCode"
              name="zip_code"
              onChange={[Function]}
              placeholder="Zip Code"
              value=""
/>,
          ],
          "className": "reportFire-form-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "for": "city",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="star"
>
                  *
</span>,
                " optional",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "*",
                  "className": "star",
                },
                "ref": null,
                "rendered": "*",
                "type": "span",
              },
              " optional",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "city",
              "name": "city",
              "onChange": [Function],
              "placeholder": "City",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "for": "state",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "state",
              "name": "state",
              "onChange": [Function],
              "placeholder": "State",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "for": "zip code",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "zipCode",
              "name": "zip_code",
              "onChange": [Function],
              "placeholder": "Zip Code",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-fire-logo"
/>,
          <h2>
            Report a Fire
</h2>,
          <section
            className="reportFire-form-section"
>
            <label
                        for="name"
            />
            <input
                        id="name"
                        name="first_name"
                        onChange={[Function]}
                        placeholder="First Name"
                        value=""
            />
            <label
                        for="lastName"
            />
            <input
                        id="lastName"
                        name="last_name"
                        onChange={[Function]}
                        placeholder="Last Name"
                        value=""
            />
</section>,
          <br />,
          <section
            className="reportFire-form-section"
>
            <h3>
                        <span
                                    className="star"
                        >
                                    *
                        </span>
                        Required
            </h3>
            <label
                        for="latitude"
            />
            <input
                        id="latitude"
                        name="latitude"
                        onChange={[Function]}
                        placeholder="Latitude"
                        required={true}
                        value=""
            />
            <label
                        for="longitude"
            />
            <input
                        id="longitude"
                        name="longitude"
                        onChange={[Function]}
                        placeholder="Longitude"
                        required={true}
                        value=""
            />
</section>,
          <br />,
          <section
            className="reportFire-form-section"
>
            <label
                        for="city"
            />
            <br />
            <h3>
                        <span
                                    className="star"
                        >
                                    *
                        </span>
                         optional
            </h3>
            <input
                        id="city"
                        name="city"
                        onChange={[Function]}
                        placeholder="City"
                        value=""
            />
            <label
                        for="state"
            />
            <input
                        id="state"
                        name="state"
                        onChange={[Function]}
                        placeholder="State"
                        value=""
            />
            <label
                        for="zip code"
            />
            <input
                        id="zipCode"
                        name="zip_code"
                        onChange={[Function]}
                        placeholder="Zip Code"
                        value=""
            />
</section>,
          <button
            onClick={[Function]}
>
            Submit
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-fire-logo",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Report a Fire",
          },
          "ref": null,
          "rendered": "Report a Fire",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                for="name"
/>,
              <input
                id="name"
                name="first_name"
                onChange={[Function]}
                placeholder="First Name"
                value=""
/>,
              <label
                for="lastName"
/>,
              <input
                id="lastName"
                name="last_name"
                onChange={[Function]}
                placeholder="Last Name"
                value=""
/>,
            ],
            "className": "reportFire-form-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "for": "name",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "name",
                "name": "first_name",
                "onChange": [Function],
                "placeholder": "First Name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "for": "lastName",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "lastName",
                "name": "last_name",
                "onChange": [Function],
                "placeholder": "Last Name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                <span
                                className="star"
                >
                                *
                </span>
                Required
</h3>,
              <label
                for="latitude"
/>,
              <input
                id="latitude"
                name="latitude"
                onChange={[Function]}
                placeholder="Latitude"
                required={true}
                value=""
/>,
              <label
                for="longitude"
/>,
              <input
                id="longitude"
                name="longitude"
                onChange={[Function]}
                placeholder="Longitude"
                required={true}
                value=""
/>,
            ],
            "className": "reportFire-form-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="star"
>
                    *
</span>,
                  "Required",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "*",
                    "className": "star",
                  },
                  "ref": null,
                  "rendered": "*",
                  "type": "span",
                },
                "Required",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "for": "latitude",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "latitude",
                "name": "latitude",
                "onChange": [Function],
                "placeholder": "Latitude",
                "required": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "for": "longitude",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "longitude",
                "name": "longitude",
                "onChange": [Function],
                "placeholder": "Longitude",
                "required": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                for="city"
/>,
              <br />,
              <h3>
                <span
                                className="star"
                >
                                *
                </span>
                 optional
</h3>,
              <input
                id="city"
                name="city"
                onChange={[Function]}
                placeholder="City"
                value=""
/>,
              <label
                for="state"
/>,
              <input
                id="state"
                name="state"
                onChange={[Function]}
                placeholder="State"
                value=""
/>,
              <label
                for="zip code"
/>,
              <input
                id="zipCode"
                name="zip_code"
                onChange={[Function]}
                placeholder="Zip Code"
                value=""
/>,
            ],
            "className": "reportFire-form-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "for": "city",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="star"
>
                    *
</span>,
                  " optional",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "*",
                    "className": "star",
                  },
                  "ref": null,
                  "rendered": "*",
                  "type": "span",
                },
                " optional",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "city",
                "name": "city",
                "onChange": [Function],
                "placeholder": "City",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "for": "state",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "state",
                "name": "state",
                "onChange": [Function],
                "placeholder": "State",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "for": "zip code",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "zipCode",
                "name": "zip_code",
                "onChange": [Function],
                "placeholder": "Zip Code",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
